agent/rpc/peering/service.go

```diff
@@ -492,13 +494,21 @@ func (s *Service) StreamResources(stream pbpeering.PeeringService_StreamResource
 		// we are not the leader so we will hang up on the dialer
 
 		s.logger.Error("cannot establish a peering stream on a follower node")
-		var merr *multierror.Error
-		if err := s.sendLeaderAddrMsg(stream, nil, s.logger); err != nil {
-			merr = multierror.Append(merr, err)
+
+		details, err := anypb.New(&pbpeering.LeaderAddress{
+			Address: s.Backend.LeaderAddress().Get(),
+		})
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("failed to marshal the leader address in response; err: %v", err))
 		}
 
-		merr = multierror.Append(grpcstatus.Error(codes.FailedPrecondition, "cannot establish a peering stream on a follower node"), merr)
-		return merr.ErrorOrNil()
+		// if there was an error above while marshaling details, then details is nil
+		// clients need to check types casts either way so it should be safe to send {nil}
+		return grpcstatus.FromProto(&spb.Status{
+			Code:    9,
+			Message: "cannot establish a peering stream on a follower node",
+			Details: []*apb.Any{details},
+		}).Err()
```

Comment:
Just realized there's a cleaner way of doing this that wasn't obvious from godocs because of this [type alias](https://pkg.go.dev/google.golang.org/grpc@v1.47.0/status#Status). The useful functions were hidden in the godocs of the internal/status.Status page.

```go
status.New(codes.FailedPrecondition, "cannot establish a peering stream on a follower node").WithDetails(&pbpeering.LeaderAddress{
	Address: s.Backend.LeaderAddress().Get(),
})
```

Reply:
thanks for the suggestion and offline clarifications!! made the change